servers/visual_server.cpp

```diff
@@ -343,6 +343,35 @@ RID VisualServer::get_white_texture() {
 #define SMALL_VEC2 Vector2(0.00001, 0.00001)
 #define SMALL_VEC3 Vector3(0.00001, 0.00001, 0.00001)
 
+// Maps normalized vector to an octohedron projected onto the cartesian plane
+// Resulting 2D vector in range [-1, 1]
+// See http://jcgt.org/published/0003/02/01/ for details
+Vector2 norm_to_oct(const Vector3 v) {
```

Comment:
Unsure where the best place to put these utility functions should be

Reply:
I think this is fine. The only other vertex function is also placed amongst the varyings and uniforms